contracts/modules/staking/StakingProducts.sol

```diff
@@ -162,12 +162,32 @@ contract StakingProducts is IStakingProducts, MasterAwareV2, Multicall {
       }
 
       if (_param.setTargetPrice) {
+
         if (_param.targetPrice > TARGET_PRICE_DENOMINATOR) {
           revert TargetPriceTooHigh();
         }
+
         if (_param.targetPrice < globalMinPriceRatio) {
           revert TargetPriceBelowMin();
         }
+
+        // if this is an existing product, when the target price is updated we need to calculate the
+        // current base price using the old target price and update the bumped price to that value
+        // uses the same logic as calculatePremium()
```

Comment:
we may want to make a pure internal function if it's the exact same logic to avoid duplicatoj